Assets/Scripts/UI/InfoWindow/InfoWindow.cs

```diff
@@ -0,0 +1,57 @@
+using Ships.Components;
+using UI.InfoWindow;
+using UnityEngine;
+
+/// <summary>
+/// Instantiates InfoWindow and attaches them to each ShipInfo in its children
+/// </summary>
+public class InfoWindow : MonoBehaviour
+{
+    [SerializeField] private SubsystemButtonManager subsystemButtonManager;
+    [SerializeField] private Camera camera;
+
+    public void Refresh(ShipInfo shipInfo)
+    {
+        if (camera != null && shipInfo != null)
+        {
+            var shipPos = shipInfo.transform.position;
+            camera.transform.position = new Vector3(shipPos.x, shipPos.y, camera.transform.position.z);
+            MatchCamera(camera, shipInfo);
+
+            subsystemButtonManager.Refresh(shipInfo);
+            gameObject.SetActive(true);
+        }
+        else
+        {
+            gameObject.SetActive(false);
+        }
+    }
+
+    private void MatchCamera(Camera cam, ShipInfo shipInfo)
+    {
+        if (cam == null || shipInfo == null)
+        {
+            return;
+        }
+
+        var position = cam.ViewportToWorldPoint(Vector3.zero);
+        var up = cam.ViewportToWorldPoint(Vector3.up) - position;
+        var right = cam.ViewportToWorldPoint(Vector3.right) - position;
+
+        var bounds = shipInfo.Data.Visuals.GetComponent<SpriteRenderer>().bounds;
+        var matchSize = Mathf.Max(bounds.size.y, bounds.size.x * up.magnitude / right.magnitude);
+
+        var multiplier = shipInfo.Data.CameraSizeMultiplier > 0 ? shipInfo.Data.CameraSizeMultiplier : 1f;
+        cam.orthographicSize = matchSize * multiplier;
```

Comment:
Don't think this will work with pixel perfect camera, since it takes control of the cameras orthographicSize. Also it should be possible to scale the camera to fit the target without needing a hard coded `CameraSizeMultiplier`.

Reply:
Hmm yeah it doesn't work.